### PR TITLE
perf(sql): parallelise sample and population covariance

### PR DESCRIPTION
**Pull request by nwoolmer**
First 1 million trades in demo trades dataset.

```
SELECT covar_samp(price, amount) from trades where symbol = 'ETH-USD'
-- gives 3.52651152818
-- takes 8-9 ms

SELECT covar_samp(price, amount) from trades where symbol = 'ETH-USD'
-- gives 3.5265115281.
-- takes 4 ms
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 9 / 28 (32.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/CovarSampleGroupByFunctionFactory.java|9|28|32.14%|

File: core/src/test/resources/sqllogictest/test/sql/test_covar.test [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+query RRRRR
+SELECT COVAR_POP(x,y), COVAR_POP(x,1), COVAR_POP(1,y), COVAR_POP(x,NULL), COVAR_POP(NULL,y) FROM integers
+----
+80.000000 0.000000 0.000000 NULL NULL
+
+
+query RRRRR
+SELECT COVAR_SAMP(x,y), COVAR_SAMP(x,1), COVAR_SAMP(1,y), COVAR_SAMP(x,NULL), COVAR_SAMP(NULL,y) FROM integers
```
Comment (ideoma) [inline core/src/test/resources/sqllogictest/test/sql/test_covar.test:68]: Usually order of parameters is (y, x), see https://docs.snowflake.com/en/sql-reference/functions/covar_samp

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 28 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/CovarSampleGroupByFunctionFactory.java|28|28|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 47 / 47 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/AbstractCovarGroupByFunction.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/CovarSampleGroupByFunctionFactory.java|28|28|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**